### DISCUSSION TITLE
HammerDB - Error in Virtual User 2 while running either virtual user test or autopilot.

### DISCUSSION THREAD

**Discussion by ronak2303** [TPC-Council/HammerDB#133] [Q&A] [answered]
i am running benchmark test setup as below its TPC-C.
SQL server is on linux machine.
running hammerdb UI on Windows server as master.

i am able to build tpcc database, one its completed with 4 virtual user.
when i run actual test with 4 virtual user under virtual user and then create and run i get below errors and only 1 user is running some script.

some times i get 
Error in Virtual User 6: can't use empty string as operand of "+"
or
Error in Virtual User 2: Connection to DRIVER=ODBC Driver 17 for SQL Server;SERVER=tcp:16001;UID=test;PWD=test could not be established : can't create object "odbc": command already exists with that name.

**Comment by sm-shaw** [comment, marked as answer]
So the first thing to check is that the database did actually get created correctly and you can connect and run SQL against it. With a Linux server and a Windows client try using the SSMS GUI to browse the schema, how many warehouses have been created? If the tables are created but the data is empty then this may give empty string type errors. There are examples of troubleshooting this type of error here https://sourceforge.net/p/hammerdb/discussion/292313/thread/6575123c/
For can't create object "odbc" are you running the GUI or the CLI (or web service?) - "odbc" is the name of the connection that it creates so we need to find why it has already created it and not destroyed it or whether another variable is already named "odbc". There are 2 separate stages - the schema build and the test the scripts are different for both. Once you run it manually - i suggest first test, then timed and then run autopilot.

**Comment by sm-shaw** [comment]
Closed as no further update
